src/core/optimization_container.jl

```diff
@@ -1,12 +1,40 @@
+struct OptimizationContainerMetadata
```

Comment:
Do we need to make this a separate struct?

Reply:
This allows us to deserialize this one object apart from the OptimizationContainer.